source/grid/grid_tools_dof_handlers.cc

```diff
@@ -2540,38 +2442,66 @@ namespace GridTools
               {
-                matching[i] = *it;
-                face2_vertices.erase(it);
+                face1_vertices[i] = *it;
+                face2_vertices[i] = i;
+                face2_vertices_set.erase(it);
                 break; // jump out of the innermost loop
               }
           }
       }
 
     // And finally, a lookup to determine the ordering bitmask:
-    if (face2_vertices.empty())
-      orientation = OrientationLookupTable<dim>::lookup(matching);
-
-    return face2_vertices.empty();
-  }
-
-
+    if (face2_vertices_set.empty())
+      {
+        const auto combined_orientation =
+          face1->reference_cell().get_combined_orientation(
+            make_array_view(face1_vertices.cbegin(),
+                            face1_vertices.cbegin() + face1->n_vertices()),
+            make_array_view(face2_vertices.cbegin(),
+                            face2_vertices.cbegin() + face2->n_vertices()));
+        std::bitset<3> orientation;
+        // The original version of this doesn't use the standardized orientation
+        // value so we have to do an additional translation step
+        if (dim == 2)
+          {
+            // In 2D, calls in set_periodicity_constraints() ultimately require
+            // calling FiniteElement::face_to_cell_index(), which in turn (for
+            // hypercubes) calls GeometryInfo<2>::child_cell_on_face(). The
+            // final function assumes that orientation in 2D is encoded solely
+            // in face_flip (the second bit) whereas the orientation bit is
+            // ignored. Hence, the backwards orientation is 1 + 2 and the
+            // standard orientation is 1 + 0.
+            constexpr std::array<unsigned int, 2> translation{{3, 1}};
+            AssertIndexRange(combined_orientation, translation.size());
+            orientation =
+              translation[std::min<unsigned int>(combined_orientation, 1u)];
+          }
+        else
+          {
+            // There are two differences between the orientation implementation
+            // used in the periodicity code and that used in ReferenceCell:
+            //
+            // 1. The bitset is unpacked as (orientation, flip, rotation)
+            //    instead of the standard (orientation, rotation, flip).
+            //
+            // 2. The 90 degree rotations are always clockwise, so the third and
+            //    seventh (in the combined orientation) are switched.
+            //
+            // Both translations are encoded in this table. This matches
+            // OrientationLookupTable<3> which was present in previous revisions
+            // of this file.
+            constexpr std::array<unsigned int, 8> translation{
+              {0, 1, 4, 7, 2, 3, 6, 5}};
+            AssertIndexRange(combined_orientation, translation.size());
+            orientation =
+              translation[std::min<unsigned int>(combined_orientation, 7u)];
```

Comment:
For sake of fun:

https://github.com/dealii/dealii/blob/47cf4a0ea98c99aba40ba86bdce1bdccbaae33d5/include/deal.II/matrix_free/face_setup_internal.h#L1005-L1025

https://github.com/dealii/dealii/blob/47cf4a0ea98c99aba40ba86bdce1bdccbaae33d5/include/deal.II/matrix_free/fe_evaluation.h#L8458-L8465

Reply:
I don't understand why you need the `std::min` call. Doesn't the assertion above already check that the value is at most 7?

Reply:
I guess it is a compile warning?

Reply:
It's a compiler warning.

Reply:
I'll clean up some of the matrix-free orientation code in a future PR (like we talked about at the workshop, I'd like to make `0` the default orientation everywhere).

Reply:
> I'll clean up some of the matrix-free orientation code in a future PR (like we talked about at the workshop, I'd like to make 0 the default orientation everywhere).

@drwells  Much appreciated!